### PR TITLE
Fix/multiple issues

### PR DESCRIPTION
**Pull request by SemraSh**

### CODE REVIEW

File: src/findUnused.test.js [hunk 1]
```diff
@@ -20,7 +22,7 @@ describe('findUnused', () => {
     expect(await findUnused(secrets)).toEqual([])
   })
   it('Returns empty array if some of the secrets are unused', async () => {
```
Comment (radomird) [inline src/findUnused.test.js:24]: The description of the test suggests that the `findUnused` returns an empty array if some of the secrets is unused, but what I see is that it's not the case. Maybe the test description should be updated?

**Review by radomird — APPROVED** [review]
LGTM

File: README.md [hunk 2]
```diff
@@ -1,28 +1,78 @@
...
+
 ```
 
 ## Usage
 
 Configure this action in your workflows providing the inputs described below.
 
+### Install a GitHub App
+
+**Required** [Create a GitHub App](https://docs.github.com/en/developers/apps/building-github-apps/creating-a-github-app) within your organisation with permission to `secrets: read-only` and [install](https://docs.github.com/en/developers/apps/managing-github-apps/installing-github-apps#installing-your-private-github-app-on-your-repository) it on the repo where you want to use this action. Save the `APP_ID` and generated `PRIVATE_KEY` to the secrets under your organisation or repo.
```
Comment (simoneb) [inline README.md:46]: nit: the right name for these permissions is "scope". let's keep the naming consistent

File: README.md [hunk 3]
```diff
@@ -1,28 +1,78 @@
...
+        id: generate_token
+        uses: tibdex/github-app-token@v1
+        with:
+          app_id: ${{ secrets.APP_ID }}
+          private_key: ${{ secrets.PRIVATE_KEY }}
+      - name: Checkout Code
+      - uses: actions/checkout@v3
       - name: Detect unused secrets
-        uses: nearform/github-action-detect-unused-secrets@v1
+        uses: nearform/github-action-detect-unused-secrets@main
```
Comment (simoneb) [inline README.md:34]: in the docs let's show using a tagged version, not a branch. basically undo this change

File: README.md [hunk 4]
```diff
@@ -1,28 +1,78 @@
 # github-action-detect-unused-secrets
 
-GitHub Action that automatically detects unused secrets comparing the workflow secret names with the repo secret names.
+GitHub Action that detects unused secrets.
+Compares the secret names in your repository with the secret names under `.github/workflows`
```
Comment (simoneb) [inline README.md:4]: let's clarify better what this does. it checks which secrets are used in workflows with those defined in the repository settings. let's also clarify very well that this action doesn't get access to the values of secrets, just their names

File: README.md [hunk 5]
```diff
@@ -1,28 +1,78 @@
...
 
 ## Example usage
 
-```yaml
-name: detect-unused-secrets
+The example below runs the action every Monday at 10am.
```
Comment (simoneb) [inline README.md:8]: it also allows triggering it manually though

File: README.md [hunk 6]
```diff
@@ -1,28 +1,78 @@
...
+Generates a token with required permissions which `secrets.GITHUB_TOKEN` does not have.
```
Comment (simoneb) [inline README.md:9]: no need to clarify this, or if you want to clarify it also clarify that secrets.GITHUB_TOKEN is the built-in token and that there's no way to give that token the necessary permissions

File: README.md [hunk 7]
```diff
@@ -1,28 +1,78 @@
...
+Uses checkout code to get access to the `$GITHUB_WORKSPACE`,
```
Comment (simoneb) [inline README.md:10]: this is an implementation detail which we don't need to clarify here. we can simply clarify that we need to check out the code so we have access to the content of the workflow files

File: README.md [hunk 8]
```diff
@@ -1,28 +1,78 @@
...
 
 ## Usage
 
 Configure this action in your workflows providing the inputs described below.
 
+### Install a GitHub App
+
+**Required** [Create a GitHub App](https://docs.github.com/en/developers/apps/building-github-apps/creating-a-github-app) within your organisation with permission to `secrets: read-only` and [install](https://docs.github.com/en/developers/apps/managing-github-apps/installing-github-apps#installing-your-private-github-app-on-your-repository) it on the repo where you want to use this action. Save the `APP_ID` and generated `PRIVATE_KEY` to the secrets under your organisation or repo.
+
+This action uses the [listRepoSecrets endpoint](https://docs.github.com/en/rest/reference/actions#list-repository-secrets), your GitHub App must have the `secrets` repository permission to use this endpoint.
```
Comment (simoneb) [inline README.md:48]: let's use the right name of this scope

File: README.md [hunk 9]
```diff
@@ -1,28 +1,78 @@
...
+
+> Note that you can skip this step if your repo is `not` under an organisation, you can [create a personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) `with repo scope` and reference it as github-token
```
Comment (simoneb) [inline README.md:50]: this is not accurate. you can use a PAT if you prefer, it doesn't matter if the repo is under an org or a personal account. it's just our decision to not use PATs for our org

File: README.md [hunk 10]
```diff
@@ -1,28 +1,78 @@
...
+  with:
+    app_id: ${{ secrets.APP_ID }}
+    private_key: ${{ secrets.PRIVATE_KEY }}
+```
+
+> Note that you can skip this step if your repo is `not` under an organisation, you can [create a personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) `with repo scope` and reference it as github-token
+
+### uses: actions/checkout@v3
+
+**Required** This action checks-out your repository under `$GITHUB_WORKSPACE`. For information check [actions/checkout](https://github.com/actions/checkout). This step is required because under the hood `detect-unused-secrets` is executing a script to grab the secret names used in `.github/workflows`.
```
Comment (simoneb) [inline README.md:72]: same comment as before, no need to explain implementation details. we just need to explain that we need access to the content of the workflow files

File: README.md [hunk 11]
```diff
@@ -1,28 +1,78 @@
...
+
 ### `github-token`
 
-**Required** A GitHub token. For additional information read this guide --> [#about-the-github_token-secret](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/authenticating-with-the-github_token#about-the-github_token-secret)
+**Required** [The generated token](#uses-tibdexgithub-app-tokenv1) or your personal access token with repo scope.
```
Comment (simoneb) [inline README.md:76]: the scope is wrong

**Review by simoneb — COMMENTED** [review]
See inline comments. Some changes to be done primarily in the docs

File: README.md (hunk 2 above)
Comment (SemraSh) [reply README.md:46]: Screenshot 2022-03-09 at 11 43 03
In this case, they are names permissions, and under the permissions tab, it might be confusing to name it "scope".

Comment (simoneb) [reply README.md:46]: you're right. the tokens have scope, the apps have permissions. https://docs.github.com/en/rest/reference/actions#list-repository-secrets

ok let's just make sure it's consistent throughout 👍

File: README.md (hunk 11 above)
Comment (SemraSh) [reply README.md:76]: The repo scope is required for this action to run with a PAT, do you mean the naming? This is named scope in Github.
Screenshot 2022-03-09 at 11 56 52

Comment (simoneb) [reply README.md:76]: Ah no you're actually right again. The repo scope is required for the PAT. Instead of specifying it in detail here, why don't we just point the users to the docs? https://docs.github.com/en/rest/reference/actions#list-repository-secrets

File: README.md [hunk 12]
```diff
@@ -1,28 +1,79 @@
...
 
 Configure this action in your workflows providing the inputs described below.
 
+### Install a GitHub App
+
+**Required** [Create a GitHub App](https://docs.github.com/en/developers/apps/building-github-apps/creating-a-github-app) within your organisation with permission to `secrets: read-only` and [install](https://docs.github.com/en/developers/apps/managing-github-apps/installing-github-apps#installing-your-private-github-app-on-your-repository) it on the repo where you want to use this action. Save the `APP_ID` and generated `PRIVATE_KEY` to the secrets under your organisation or repo.
+
+This action uses the [listRepoSecrets endpoint](https://docs.github.com/en/rest/reference/actions#list-repository-secrets), your GitHub App must have the `secrets` repository permission to use this endpoint.
+
+> Note that you can skip this step if you use a [personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) `with repo scope` and reference it as github-token
```
Comment (simoneb) [inline README.md:51]:
```suggestion
> Note that you can skip this step if you use a [personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) with `repo` scope and reference it as github-token
```

**Review by simoneb — APPROVED** [review]

**Review by radomird — APPROVED** [review]
